toolchain/sem_ir/inst.h

```diff
@@ -431,11 +435,31 @@ class InstStore {
     return TryGetAs<InstT>(inst_id);
   }
 
-  auto GetLocId(InstId inst_id) const -> LocId {
+  // Returns a resolved LocId, which will point to a parse node, an import, or
+  // be None.
+  //
+  // Unresolved LocIds can be backed by an InstId which may or may not have a
+  // value after being resolved, so this operation needs to be done before using
+  // most operations on `LocId`.
+  auto GetResolvedLocId(LocId loc_id) const -> LocId {
+    while (loc_id.kind() == SemIR::LocId::Kind::InstId) {
+      auto inst_id = loc_id.inst_id();
+      CARBON_CHECK(inst_id.index >= 0, "{0}", inst_id.index);
+      CARBON_CHECK(inst_id.index < (int)loc_ids_.size(), "{0} {1}",
```

Comment:
We should be doing static_cast<int>(...), not C-style casts. I see the pre-existing copy below, but we probably shouldn't copy it.

```suggestion
      CARBON_CHECK(inst_id.index < static_cast<int>(loc_ids_.size()), "{0} {1}",
```

Reply:
Done, but casting the index to size_t instead. We've checked it's >= 0 on the line above so that cast isn't lossy. Casting to an int could be.